src/lua/api-gateway/validation/oauth2/oauthTokenValidator.lua

```diff
@@ -198,7 +215,7 @@ function _M:validateOAuthToken()
 
     if (cachedToken ~= nil) then
         -- ngx.log(ngx.INFO, "Cached token=" .. cachedToken)
-        local obj = cjson.decode(cachedToken)
+        local obj =  self:safeDecode(cachedToken)
         local tokenValidity, error = self:isCachedTokenValid(obj)
```

Comment:
isCachedTokenValid does not do a check if **obj** is nil. If the token cannot be deserialised then the lua code will throw an exception inside isCachedTokenValid